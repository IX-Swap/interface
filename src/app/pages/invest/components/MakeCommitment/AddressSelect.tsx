import React from 'react'
import { useWithdrawalAddresses } from 'app/pages/accounts/pages/withdrawalAddresses/hooks/useWithdrawalAddresses'
import { TypedField } from 'components/form/TypedField'
import { WithdrawalAddressSelect } from 'components/form/WithdrawalAddressSelect'
import { useFormContext } from 'react-hook-form'

export const AddressSelect = () => {
  const { control } = useFormContext()
  const { data, status } = useWithdrawalAddresses({})
  const filteredAddresses = data.list.filter(
    ({ status }) => status === 'Approved'
  )
  const hasFilteredAddresses = filteredAddresses.length > 0

  return (
    <TypedField
      component={WithdrawalAddressSelect}
      control={control}
      name='withdrawalAddress'
      label='Destination Wallet Address'
      placeholder={
        !hasFilteredAddresses && status !== 'loading'
          ? 'You do not have wallet addresses'
          : 'Select walet'
      }
      disabled={!hasFilteredAddresses}
      displayEmpty
      list={filteredAddresses}
      status={status}
    />
  )
}
