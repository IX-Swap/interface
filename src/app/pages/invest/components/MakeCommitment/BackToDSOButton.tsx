import React from 'react'
import { Button, ButtonProps, Box } from '@mui/material'
import { generatePath, useParams } from 'react-router-dom'
import { Icon } from 'ui/Icons/Icon'
import { useAppBreakpoints } from 'hooks/useAppBreakpoints'
import { InvestRoute } from 'app/pages/invest/router/config'

export const BackToDSOButton = (props: ButtonProps) => {
  const { isMobile } = useAppBreakpoints()
  const { dsoId, issuerId } = useParams<{ dsoId: string; issuerId: string }>()

  return (
    <Box height='100%' display='flex' alignItems='center'>
      <Button
        {...props}
        endIcon={isMobile ? undefined : <Icon name='arrow-right' />}
        startIcon={isMobile ? <Icon name='arrow-left' /> : undefined}
        href={generatePath(InvestRoute.view, { dsoId, issuerId })}
        sx={{
          pl: {
            xs: 0,
            md: 4
          }
        }}
      >
        Back to DSO page
      </Button>
    </Box>
  )
}
