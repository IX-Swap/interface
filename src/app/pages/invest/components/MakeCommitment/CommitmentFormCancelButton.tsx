import React from 'react'
import { Button, ButtonProps } from '@mui/material'
import { AppRouterLinkComponent } from 'components/AppRouterLink'
import { useParams } from 'react-router-dom'
import { InvestRoute } from 'app/pages/invest/router/config'

export interface CommitmentFormCancelButtonProps extends ButtonProps {}

export const CommitmentFormCancelButton = (
  props: CommitmentFormCancelButtonProps
) => {
  const params = useParams<{ dsoId: string; issuerId: string }>()

  return (
    <Button
      {...props}
      variant='outlined'
      fullWidth
      color='primary'
      disableElevation
      component={AppRouterLinkComponent}
      to={InvestRoute.view}
      params={params}
      replace
    >
      Cancel
    </Button>
  )
}
