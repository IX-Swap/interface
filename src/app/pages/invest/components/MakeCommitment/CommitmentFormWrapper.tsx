import React, { PropsWithChildren } from 'react'
import { CommitmentFormValues } from 'types/commitment'
import { FormProps, Form } from 'components/form/Form'
import { useMakeCommitment } from 'app/pages/invest/hooks/useMakeCommitment'
import {
  commitmentCampaignValidationSchema,
  commitmentNonCampaignValidationSchema
} from 'app/pages/invest/validation'

export interface CommitmentFormProps {
  dso: string
  currency: string
  isCampaign?: boolean
}

export const CommitmentFormWrapper = (
  props: PropsWithChildren<
    CommitmentFormProps & FormProps<CommitmentFormValues>
  >
) => {
  const { dso, currency, children, isCampaign = false, ...rest } = props
  const validationSchema = isCampaign
    ? commitmentCampaignValidationSchema
    : commitmentNonCampaignValidationSchema
  const {
    invest: [makeInvestment]
  } = useMakeCommitment()
  const handleSubmit = async ({
    totalAmount,
    pricePerUnit,
    withdrawalAddress,
    ...values
  }: CommitmentFormValues) => {
    await makeInvestment({
      ...values,
      withdrawalAddress:
        withdrawalAddress === '' ? undefined : withdrawalAddress,
      signedSubscriptionDocument: values.signedSubscriptionDocument?._id,
      dso,
      currency
    })
  }

  return (
    <Form {...rest} onSubmit={handleSubmit} validationSchema={validationSchema}>
      {children}
    </Form>
  )
}
