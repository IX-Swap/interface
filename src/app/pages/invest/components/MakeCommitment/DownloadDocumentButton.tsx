import { DownloadDSOSubscriptionDocument } from 'app/components/DSO/components/DownloadDSOSubscriptionDocument'
import React from 'react'
import { DigitalSecurityOffering } from 'types/dso'
import { Icon } from 'ui/Icons/Icon'

export interface DownloadDocumentButtonProps {
  dso: DigitalSecurityOffering
}

export const DownloadDocumentButton = ({
  dso
}: DownloadDocumentButtonProps) => {
  const isCampaign = dso.isCampaign === true
  const downloadButton = isCampaign
    ? 'Download Investment Agreement'
    : 'Download Subscription Document'

  return (
    <DownloadDSOSubscriptionDocument
      dsoId={dso._id}
      name={dso.subscriptionDocument?.originalFileName}
      fullWidth
      variant='outlined'
      startIcon={<Icon name='download' />}
      sx={{
        borderStyle: 'dashed'
      }}
    >
      {downloadButton}
    </DownloadDSOSubscriptionDocument>
  )
}
