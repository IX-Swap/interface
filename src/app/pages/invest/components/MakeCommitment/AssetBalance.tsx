import React from 'react'
import { formatMoney } from 'helpers/numbers'
import { OverviewValue } from 'app/pages/invest/components/MakeCommitment/OverviewValue'
import { Grid } from '@mui/material'
import { NoBalance } from 'app/pages/invest/components/MakeCommitment/NoBalance'
import { useCurrencyBalance } from 'app/pages/invest/hooks/useCurrencyBalance'

export interface AssetBalanceProps {
  symbol: string
}

export const AssetBalance = ({ symbol }: AssetBalanceProps) => {
  const currencyBalance: number = useCurrencyBalance(symbol)

  return (
    <Grid container spacing={3}>
      <Grid item xs={12}>
        <OverviewValue
          label='Available Balance'
          value={formatMoney(currencyBalance, symbol)}
        />
      </Grid>
      {currencyBalance <= 0 && (
        <Grid item xs={12}>
          <NoBalance />
        </Grid>
      )}
    </Grid>
  )
}
