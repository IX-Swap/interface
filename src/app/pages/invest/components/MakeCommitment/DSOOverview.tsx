import { Paper, Grid } from '@mui/material'
import { DSOTitle } from 'app/components/DSO/components/DSOTitle'
import { AssetBalance } from 'app/pages/invest/components/MakeCommitment/AssetBalance'
import { DSOBlockchainDetails } from 'app/pages/invest/components/DSOBlockChainDetails/DSOBlockchainDetails'
import { OverviewValue } from 'app/pages/invest/components/MakeCommitment/OverviewValue'
import React from 'react'
import { DigitalSecurityOffering } from 'types/dso'

export interface DSOOverviewProps {
  dso: DigitalSecurityOffering
}

export const DSOOverview = ({ dso }: DSOOverviewProps) => {
  return (
    <Paper sx={{ p: { xs: 2, md: 3 } }}>
      <Grid container spacing={3}>
        <Grid item xs={12}>
          <DSOTitle dso={dso} />
        </Grid>
        <Grid item xs={12}>
          <OverviewValue
            label='Minimum Investment'
            value={`${dso.minimumInvestment ?? 0} ${dso.tokenSymbol}`}
          />
        </Grid>
        <Grid item xs={12}>
          <AssetBalance symbol={dso.currency.symbol} />
        </Grid>
        <Grid item xs={12}>
          <DSOBlockchainDetails dso={dso} />
        </Grid>
      </Grid>
    </Paper>
  )
}
