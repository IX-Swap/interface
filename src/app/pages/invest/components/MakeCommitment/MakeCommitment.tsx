import React from 'react'
import { useDSOById } from 'app/pages/invest/hooks/useDSOById'
import { Grid } from '@mui/material'
import { useParams } from 'react-router-dom'
import { useCommitmentActivity } from 'app/pages/invest/hooks/useCommitmentActivity'
import { DSOOverview } from 'app/pages/invest/components/MakeCommitment/DSOOverview'
import { RootContainer } from 'ui/RootContainer'

export const MakeCommitment = () => {
  const params = useParams<{ dsoId: string; issuerId: string }>()
  const { data, isLoading } = useDSOById(params.dsoId, params.issuerId)

  useCommitmentActivity(data?._id)

  if (isLoading || data === undefined) {
    return null
  }

  return (
    <RootContainer>
      <Grid container spacing={3}>
        <Grid item xs={12} md={4}>
          <DSOOverview dso={data} />
        </Grid>
        <Grid item xs={12} md={8}></Grid>
      </Grid>
    </RootContainer>
  )
}
