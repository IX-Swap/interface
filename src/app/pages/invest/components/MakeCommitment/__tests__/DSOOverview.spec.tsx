import { DSOOverview } from 'app/pages/invest/components/MakeCommitment/DSOOverview'
import React from 'react'
import { render, cleanup } from 'test-utils'
import { dso } from '__fixtures__/authorizer'

jest.mock(
  'app/pages/invest/components/DSOBlockChainDetails/DSOBlockchainDetails',
  () => ({
    DSOBlockchainDetails: jest.fn(() => null)
  })
)

describe('DSOOverview', () => {
  afterEach(async () => {
    await cleanup()
    jest.clearAllMocks()
  })

  it('renders without errors', () => {
    render(<DSOOverview dso={dso} />)
  })

  it('should match snapshot', () => {
    const { container } = render(<DSOOverview dso={dso} />)

    expect(container).toMatchSnapshot()
  })
})
