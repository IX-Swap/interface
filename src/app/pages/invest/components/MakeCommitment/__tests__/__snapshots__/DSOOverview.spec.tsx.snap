// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DSOOverview should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-q29v6p-MuiPaper-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-ofrcno-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-69x3yt-MuiAvatar-root"
              style="width: 60px; height: 60px;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-1ql7le6-MuiSvgIcon-root-MuiAvatar-fallback"
                data-testid="PersonIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 css-vxrdkt-MuiTypography-root"
            >
              token name
            </h4>
            <p
              class="MuiTypography-root MuiTypography-body1 css-1b4l2pg-MuiTypography-root"
            >
              issuer name
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 css-3phns2-MuiTypography-root"
            >
              Minimum Investment
            </h6>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 css-1b4l2pg-MuiTypography-root"
            >
              100 token symbol
            </h6>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1 css-3phns2-MuiTypography-root"
                >
                  Available Balance
                </h6>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1 css-1b4l2pg-MuiTypography-root"
                >
                  SGD 0.00
                </h6>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-199li38-MuiPaper-root-MuiCard-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1 MuiGrid-grid-xs-12 css-1ie9my4-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ek9rqb-MuiSvgIcon-root"
                      data-testid="MonetizationOnOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm.31-8.86c-1.77-.45-2.34-.94-2.34-1.67 0-.84.79-1.43 2.1-1.43 1.38 0 1.9.66 1.94 1.64h1.71c-.05-1.34-.87-2.57-2.49-2.97V5H10.9v1.69c-1.51.32-2.72 1.3-2.72 2.81 0 1.79 1.49 2.69 3.66 3.21 1.95.46 2.34 1.15 2.34 1.87 0 .53-.39 1.39-2.1 1.39-1.6 0-2.23-.72-2.32-1.64H8.04c.1 1.7 1.36 2.66 2.86 2.97V19h2.34v-1.67c1.52-.29 2.72-1.16 2.73-2.77-.01-2.2-1.9-2.96-3.66-3.42z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <h6
                      class="MuiTypography-root MuiTypography-subtitle2 css-buvp3o-MuiTypography-root"
                    >
                      You don't have enough money
                    </h6>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-3phns2-MuiTypography-root"
                  >
                    Insufficient balance, please top-up your account
                  </p>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
                >
                  <a
                    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-xpk4k5-MuiButtonBase-root-MuiButton-root"
                    href="/app/accounts/cash-deposits"
                    tabindex="0"
                  >
                    Top-up my account
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
      />
    </div>
  </div>
</div>
`;
